resources/assets/coffee/react/_components/beatmapset-panel.coffee

```diff
@@ -50,16 +50,27 @@ class @BeatmapsetPanel extends React.PureComponent
     # arbitrary number
     maxDisplayedDifficulty = 10
 
+    condenseDifficulties = beatmapset.beatmaps.length > maxDisplayedDifficulty
+
+    beatmapIcon = (beatmap, showTitle) ->
+      div
+        className: "#{'beatmapset-panel__difficulty-icon' if !condenseDifficulties}"
+        key: beatmap.id
+        el BeatmapIcon, beatmap: beatmap, showTitle: showTitle
+
     difficulties =
-      for own _mode, beatmaps of BeatmapHelper.group beatmapset.beatmaps[..maxDisplayedDifficulty - 1]
-        for b in beatmaps
-          div
-            className: 'beatmapset-panel__difficulty-icon'
-            key: b.id
-            el BeatmapIcon, beatmap: b
-
-    if beatmapset.beatmaps.length > maxDisplayedDifficulty
-      difficulties.push span key: 'over', "+#{(beatmapset.beatmaps.length - maxDisplayedDifficulty)}"
+      for own mode, beatmaps of BeatmapHelper.group beatmapset.beatmaps
```

Comment:
a combination of previous span and this is slightly better, I think. Also, `hardestDiff` can be inlined with `_.last(beatmaps)`.

```coffeescript
if condenseDifficulties
  [
    el BeatmapIcon, key: "#{mode}-icon", beatmap: _.last(beatmaps), showTitle: false
    span
      className: 'beatmapset-panel__difficulty-count'
      key: "#{(mode)}-count", beatmaps.length
  ]
else
  for b in beatmaps
    div
      className: 'beatmapset-panel__difficulty-icon'
      key: b.id
      el BeatmapIcon, beatmap: b
```